File: python/raindrops/raindrops.py
from collections import OrderedDict


def convert(number):
    sounds = OrderedDict([[3, 'Pling'], [5, 'Plang'], [7, 'Plong']])

    return ''.join([sound for (factor, sound) in sounds.items() if number % factor == 0]) or str(number)

File: go/raindrops/raindrops.go
// Package raindrops provides utility functions to manipulate raindrops.
package raindrops

import "strconv"

// Convert returns a string representation of a number according to the "raindrops" rules.
func Convert(n int) string {
	s := ""

	if n%3 == 0 {
		s += "Pling"
	}
	if n%5 == 0 {
		s += "Plang"
	}
	if n%7 == 0 {
		s += "Plong"
	}

	if s == "" {
		s = strconv.Itoa(n)
	}

	return s
}
